runtime/reconcilers/canvas.go

```diff
@@ -122,3 +129,77 @@ func (r *CanvasReconciler) checkAnyComponentHasValidSpec(ctx context.Context, re
 	}
 	return false, nil
 }
+
+// validateMetricsViewTimeConsistency checks that all the metrics views referenced by the canvas' components have the same first_day_of_week and first_month_of_year.
+func (r *CanvasReconciler) validateMetricsViewTimeConsistency(ctx context.Context, refs []*runtimev1.ResourceName) error {
+	metricsViews := make(map[string]*runtimev1.Resource)
+	for _, ref := range refs {
+		// Skip non-component refs
+		if ref.Kind != runtime.ResourceKindComponent {
+			continue
+		}
+		component, err := r.C.Get(ctx, ref, false)
+		if err != nil {
+			continue
+		}
```

Comment:
nit: ensures `ctx` cancellations propagate quickly

```suggestion
		component, err := r.C.Get(ctx, ref, false)
		if err != nil {
			if errors.Is(err, drivers.ErrResourceNotFound) {
				continue
			}
			return err
		}
```